.github/workflows/deploy.yml

```diff
@@ -34,6 +35,7 @@ jobs:
     # Deploy the site built into the gh-pages branch.
     # You need to configure pages to be at gh-pages and serve the root folder.
     - name: Deploy
+      if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' }}
```

Comment:
This conditional allows us to run builds on each PR, validating the site builds, while still only deploying when the push to master goes through.